homeassistant/helpers/template.py

```diff
@@ -1336,6 +1336,44 @@ def integration_entities(hass: HomeAssistant, entry_name: str) -> Iterable[str]:
     ]
 
 
+def integration_devices(hass: HomeAssistant, entry_name: str) -> Iterable[str]:
+    """Get device ids for device tied to an integration/domain.
+
+    Provide entry_name as domain to get all device id's for a integration/domain
+    or provide a config entry title for filtering between instances of the same
+    integration.
+    """
+
+    # Don't allow searching for config entries without title
+    if not entry_name:
+        return []
+
+    # first try if there are any config entries with a matching title
+    devices: list[str] = []
+    dev_reg = device_registry.async_get(hass)
+    for entry in hass.config_entries.async_entries():
```

Comment:
This code is very inefficient

Reply:
integration_entities is frequently at the top of template profiles I see from users. I think we should avoid repeating that pattern